### PR TITLE
fix: remove dependency on github.com/pkg/errors

### LINKED ISSUE #113627: Remove dependency on github.com/pkg/errors
**Why is this needed?**

The functionality that the package provides is already present in golang's builtin errors (using `%w` in `fmt.Errof`) and the github repo is now read-only.

/kind cleanup
/priority backlog
/good-first-issue

**Notes to people that want to collaborate**

Avoid creating a single PR for the entire repo. Create multiple PRs, each targeting a specific SIG.

Do **not** submit PRs for `cmd/kubeadm`

### PR DESCRIPTION
**Pull request by jbiers**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Removes a deprecated dependency

#### Which issue(s) this PR fixes:
Fixes #113627

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go [hunk 1]
```diff
@@ -22,8 +22,6 @@ import (
 	"io"
 	"time"
 
-	"github.com/pkg/errors"
-
 	"github.com/jonboulle/clockwork"
```
Comment (lauchokyip) [inline staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go:25]: nit: I believe the format is
```
go packages

third party packages

k8s packages
```

so you need a space here

**Comment by jbiers** [comment]
@lauchokyip @dims

**Pull request closed without merging**